tango_ros_common/tango_ros_native/src/tango_ros_node.cpp

```diff
@@ -974,67 +983,82 @@ void TangoRosNode::PublishMeshMarker() {
         t3dr_image.data = last_color_image_buffer->data;
         // Get updated mesh segment indices.
         Tango3DR_Status result =
-            Tango3DR_update(t3dr_context_, &t3dr_depth, &last_camera_depth_pose_, &t3dr_image,
+            Tango3DR_updateFromPointCloud(t3dr_context_, &t3dr_depth, &last_camera_depth_pose_, &t3dr_image,
                             &last_camera_color_pose_, &t3dr_updated_indices);
         if (result != TANGO_3DR_SUCCESS) {
-          LOG(ERROR) << "Tango3DR_update failed with error code " << result;
+          LOG(ERROR) << "Tango3DR_updateFromPointCloud failed with error code: " << result;
         }
       }
-      if (t3dr_updated_indices == nullptr) {
-        LOG(INFO) << "No indices updated in mesh";
-        continue;
-      }
-      visualization_msgs::MarkerArray mesh_marker_array;
-      for (size_t i = 0; i < t3dr_updated_indices->num_indices; ++i) {
-        // Extract tango mesh from updated index.
-        Tango3DR_Mesh* tango_mesh;
-        if(Tango3DR_extractMeshSegment(
-            t3dr_context_, t3dr_updated_indices->indices[i], &tango_mesh) != TANGO_3DR_SUCCESS) {
-          LOG(ERROR) << "Tango3DR_extractMeshSegment failed";
+      // Publish Tango mesh as visualization marker.
+      if (mesh_marker_publisher_.getNumSubscribers() > 0) {
+        visualization_msgs::MarkerArray mesh_marker_array;
+        for (size_t i = 0; i < t3dr_updated_indices.num_indices; ++i) {
```

Comment:
I think we should try to keep the extract functions out of the conversion functions, I'm not sure if that would work with your suggestion.